Std/Data/Nat/Basic.lean

```diff
@@ -15,3 +15,19 @@ instance : Dvd Nat := ⟨fun a b => ∃ c, b = a * c⟩
 
 /-- Sum of a list of natural numbers. -/
 protected def sum (l : List Nat) : Nat := l.foldr (·+·) 0
+
+/--
+Integer square root function. Implemented via Newton's method.
+-/
+def sqrt (n : Nat) : Nat :=
+  let guess := n / 2
+  if guess = 0 then n else
+  let rec iter (guess : Nat) : Nat :=
+    let next := (guess + n / guess) / 2
```

Comment:
```suggestion
    let next := (guess + n / guess) >>> 1
```
Would be nice if the compiler automatically did strength reduction but it doesn't seem to looking at the IR.

Reply:
I'll add that to the new code generator when the LetExpr refactoring is done.